Optimize CSI and Flexvolume test cases

TehTotalPwnage opened the pull request:
This PR helps clean up some of the clutter between the `test_csi.py` and `test_flexvolume.py` test cases:
- Since both use essentially the same test when checking if the provisioner (for the StorageClass) works properly, those tests have been merged into `test_provisioner.py`.
- `test_flexvolume.py` retains its original tests which were specific to the Flexvolume driver.
- Common functions that are now used by both `test_provisioner.py` and `test_flexvolume.py` have now been moved into `common.py`.
- Due to the lack of tests in `test_csi.py` at the moment, that test file has been deleted. The file can be recreated when CSI specific tests are written for it.
- The test will automatically select and deselect CSI and Flexvolume tests when running based on whether or not the `csi-provisioner-0` pod exists.

This PR resolves #61 as well as resolves #65.

yasker commented:
CSI also needs the static test similar to the Flexvolume. See https://kubernetes.io/docs/concepts/storage/volumes/ search for `CSI`.

yasker commented:
Feel like we should able to merge everything including `test_provisioner.py` to `test_driver.py`. All the code would be very similar except for the static field name of Flexvolume is `flexVolume` vs `csi`, `options` vs `volumeAttributes`. Just choose Flexvolume or CSI depends on which one was detected.

TehTotalPwnage commented:
Actually, wouldn't a CSI volume need to be defined as a PersistentVolume and PersistentVolumeClaim, then attached to a test Pod? According to the Kubernetes documentation [here](https://kubernetes.io/docs/concepts/storage/volumes/#out-of-tree-volume-plugins):
> The csi volume type does not support direct reference from Pod and may only be referenced in a Pod via a PersistentVolumeClaim object.

My testing seems to confirm this: trying to use `example.yaml` from `longhorn-manager` with `flexVolume` replaced with `csi` throws this error: 
```
error: error validating "example/example-test.yaml": error validating data: ValidationError(Pod.spec.volumes[0]): unknown field "csi" in io.k8s.api.core.v1.Volume; if you choose to ignore these errors, turn validation off with --validate=false
```

yasker commented:
@TehTotalPwnage You're right. But CSI does have the document on the static allocated volume, so it should be able to be used to create a PV? So the PV is created manually rather than dynamically created by the provisioner.

I cannot get our CSI working with PV at the moment though. Maybe there is a bug.

TehTotalPwnage commented:
Commit history is cleaned up, although there might still be room for improvements in regards to squashing some of the commits.

yasker commented on manager/integration/tests/test_csi.py:
```diff
@@ -0,0 +1,156 @@
+@pytest.mark.csi  # NOQA
+def test_csi_mount(clients, core_api, volume_name): # NOQA
+    """
+    Test that a statically defined CIS volume can be created, mounted,
```
`CSI`

yasker commented on manager/integration/tests/test_csi.py:
```diff
@@ -0,0 +1,156 @@
+    pod_name = 'csi-mount-test'
+    volume_size = DEFAULT_VOLUME_SIZE * Gi
+    # Creating a volume through Longhorn requires the raw size, not the pretty
+    # Kubernetes string.
```
Really? Use e.g. `3Gi` should be fine. Sounds like a bug to me.

TehTotalPwnage commented:
Fixed the typo in `test_csi.py`. I wasn't aware that you could use Kubernetes strings based on the tests in `test_basic.py`, so I've updated those two.

I don't have a Kubernetes v1.9 cluster but I can spin one up to test Flexvolume.

yasker commented:
LGTM. Merged, thanks.
